operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/EventSource.java

```diff
@@ -21,6 +21,7 @@
   void setEventHandler(EventHandler handler);
 
   static String defaultNameFor(EventSource source) {
-    return source.getClass().getCanonicalName();
+    // we can have multiple event sources for the same class
+    return source.getClass().getName() + "#" + source.hashCode();
```

Comment:
🤔 I'm not sure how this would work… Why do we need a name if the generated names are unique already? Also, this only works if the user already has access to the `EventSource` instance, which kind of defeats the purpose of being able to retrieve it by its name… so something doesn't really work here.

Reply:
The previous implementation simply does not work when there are multiple instances with the same type. Those would get the same name (there was already a bug with that) so basically not addressable. The default name is only for the case when the user does not want to explicitly address the event source. 
So either:
- there is just 1 event source by type, so the resource is accessed via the context and type
- the cached resource addressed via a dependent resource

On every other cases it's just needs an explicit name, so the generated name is only basically a workaround to user does not need to think about a name is it's not used anywhere. It's correct in this sense.

Reply:
Note that InformerEventSoure for a config map and a deployment for example still has the same class.

Reply:
What could ne done is just move this to EventSourceInitializer method and make it more explicit that is generated. Would make ti a separate PR.

Reply:
ok pushed a small refactor regarding to this